### PR TITLE
Fixed MSVC compile warnings

### PR DESCRIPTION
**Pull request by tkoolen**

### CODE REVIEW

File: systems/plants/RigidBodyManipulator.cpp [hunk 1]
```diff
@@ -248,7 +248,7 @@ void getFiniteIndexes(T const & v, std::vector<int> &finite_indexes)
 {
   finite_indexes.clear();
   const size_t n = v.size();
-  for (size_t x = 0; x < n; x++)
+  for (int x = 0; x < n; x++)
```
Comment (psiorx) [inline systems/plants/RigidBodyManipulator.cpp]: Does MSVC not like initializing size_t in for loops?  The loop condition is comparing to a size_t.

Comment (tkoolen) [inline systems/plants/RigidBodyManipulator.cpp]: the problem is pushing x onto finite_indexes, which is a vector<int>.

Comment (psiorx) [inline systems/plants/RigidBodyManipulator.cpp]: Just realized this is my code.  It probably should have been a vector of size_t, looking back at it.

**Comment by psiorx** [comment]
Looks like a merge-conflict got checked in before resolution.

**Comment by tkoolen** [comment]
Rebased onto master.

**Comment by tkoolen** [comment]
@psiorx added some additional fixes.

File: systems/plants/constraint/testSingleTimeLinearPostureConstraintmex.cpp [hunk 2]
```diff
@@ -67,10 +67,10 @@ void mexFunction(int nlhs,mxArray* plhs[], int nrhs, const mxArray * prhs[])
     retvec_size = 1;
   }
   plhs[2] = mxCreateDoubleMatrix(num_cnst,retvec_size,mxREAL);
-  memcpy(mxGetPrSafe(plhs[2]),c.data(),sizeof(double)*num_cnst);
-  plhs[3] = mxCreateDoubleMatrix(iAfun.size(),retvec_size,mxREAL);
-  plhs[4] = mxCreateDoubleMatrix(jAvar.size(),retvec_size,mxREAL);
-  plhs[5] = mxCreateDoubleMatrix(A.size(),retvec_size,mxREAL);
+  memcpy(mxGetPr(plhs[2]),c.data(),sizeof(double)*num_cnst);
```
Comment (RussTedrake) [inline systems/plants/constraint/testSingleTimeLinearPostureConstraintmex.cpp]: why is this reverting back from mexGetPrSafe?

Comment (tkoolen) [inline systems/plants/constraint/testSingleTimeLinearPostureConstraintmex.cpp]: good catch.

**Pull request merged**